manager/allocator/network/driver/driver.go

```diff
@@ -0,0 +1,259 @@
+package driver
+
+import (
+	"net"
+
+	"github.com/docker/docker/pkg/plugingetter"
+	"github.com/docker/libnetwork/datastore"
+	"github.com/docker/libnetwork/driverapi"
+
+	"github.com/docker/swarmkit/api"
+	"github.com/docker/swarmkit/manager/allocator/network/errors"
+)
+
+const (
+	// DefaultDriver defines the name of the driver to be used by
+	// default if a network without any driver name specified is
+	// created.
+	DefaultDriver = "overlay"
+)
+
+// DrvRegistry is an interface implementing the subset of methods from
+// DrvRegistry that we actually care about. It exists to make testing this
+// Allocator easier, by breaking most of the dependency with the complexity of
+// DrvRegistry
+//
+// DrvRegistry (the real one, from libnetwork) is a tool that manages internal
+// and external drivers. 99% of its use to us is the fact that it keeps track
+// of available drivers, and that's it. The remaining 1% is the "remote"
+// driver. That particular kind of driver has an Init function that registers a
+// handler with a PluginGetter, which in turn contains a callback to the
+// drvRegistry, which, as far as I can tell, causes newly added plugins to be
```

Comment:
nitpick: `drvRegistry` -> `DrvRegistry` here and in the line below?  It seems like you're referring to the particular type or an instance of that type, as opposed to the specific private field in `allocator`.